src/lightning/pytorch/cli.py

```diff
@@ -576,12 +586,14 @@ def _parse_ckpt_path(self) -> None:
                     "class_path": hparams.pop("_class_path"),
                     "dict_kwargs": hparams,
                 }
-            hparams = {self.config.subcommand: {"model": hparams}}
+            hparams = {"model": hparams}
             try:
-                self.config = self.parser.parse_object(hparams, self.config)
-            except SystemExit:
+                subparser = parser._subcommands_action._name_parser_map[self.config.subcommand]
+                subparser.set_defaults(hparams)
+            except KeyError:
                 sys.stderr.write("Parsing of ckpt_path hyperparameters failed!\n")
                 raise
+            self.parse_arguments(parser, args)
```

Comment:
Like this the entire parsing is done twice. The speed is not a problem, since training/predicting is normally significantly slower than the parsing. However, there might be some edge case where the first parsing fails, even though this second one would succeed. But, I am fine with leaving it like this. If there are problematic edge cases, then we can fix them when they happen.